include/picongpu/plugins/Emittance.hpp

```diff
@@ -869,35 +851,22 @@ namespace picongpu
                 0,
                 commGather
             ));
-            if (gatherRank ==0){
-                std::cout << "y_offsets if gatherRank = 0: ";
-                for (int i = 0; i < gatherSize; i++){
-                    std::cout << y_offsets.at(i) << " ";
-                }
-                std::cout << std::endl;
-            }
-            if (gatherRank ==0){
-                std::cout << "globalCount_e.getDataPointer( ) if gatherRank = 0: ";
-                for (int i = 0; i < subGrid.getGlobalDomain().size.y(); i++){
-                    std::cout << globalCount_e.getDataPointer( )[i] << " ";
-                }
-                std::cout << std::endl;
-            }
+
             /* print timestep, emittance to file: */
             if( writeToFile )
             {
                 using dbl = std::numeric_limits< float_64 >;
                 outFile.precision( dbl::digits10 );
-                if (currentStep > 0){
+                if (currentStep > 0.0){
                     int startWindow_y = movingWindow.globalDimensions.offset.y();
                     int endWindow_y = movingWindow.globalDimensions.size.y() + startWindow_y;
                     if (fisttimestep == true) 
                     {
-                        outFile << "#step 0 " ; // #step, emit_all
-                        for (int i = startWindow_y; i < endWindow_y; i+=10)
+                        outFile << "#step emit_all " << std::scientific;
+                        for (int i = startWindow_y; i < (endWindow_y + 10); i+=10)
                         {
                             outFile << i * UNIT_LENGTH * SI::CELL_WIDTH_SI;
-                            if( i != endWindow_y - 1 )
+                            if( i < (endWindow_y) )
```

Comment:
This should be `if( i < ( endWindow_y - 10 - 1 )  )` because there is no need for a space after the last element.
Never the less if you remove the space in line [865](https://github.com/ComputationalRadiationPhysics/picongpu/pull/2588/files/fbd74f4e7ce6c92210d630073cf0764bc7d10986..c45480b77eb41a6bcac688badc044268eb1e8e2f#diff-f1fb3cee53e1f5187bf099a126e22114R865) you can write

```C++
outFile << "  "<< i * UNIT_LENGTH * SI::CELL_WIDTH_SI;
```